### PR TITLE
migrate to kubetest2

### PR DESCRIPTION
**Pull request by leiyiz**
**What type of PR is this?**
 /kind failing-test

**What this PR does / why we need it**:
the managed driver testgrid is failing and we suspect it's due to some issue that's only fixed in kubetest2

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/k8s-integration/main.go [hunk 1]
```diff
@@ -55,10 +56,11 @@ var (
 	saFile            = flag.String("service-account-file", "", "path of service account file")
 	deployOverlayName = flag.String("deploy-overlay-name", "", "which kustomize overlay to deploy the driver with")
 	doDriverBuild     = flag.Bool("do-driver-build", true, "building the driver from source")
-	useStagingDriver  = flag.Bool("use-staging-driver", false, "use GKE managed Filestore CSI driver for the tests")
+	useManagedDriver  = flag.Bool("use-gke-driver", false, "use GKE managed Filestore CSI driver for the tests")
 
 	// Test flags
 	testFocus = flag.String("test-focus", "External.Storage", "test focus for Kubernetes e2e")
+	parallel  = flag.Int("parallel", 4, "the number of parallel tests setting for ginkgo parallelism")
```
Comment (saikat-royc) [inline test/k8s-integration/main.go:63]: let's keep it the same as current which is 3

File: test/run-k8s-integration.sh [hunk 2]
```diff
@@ -26,19 +26,25 @@ readonly gke_release_channel=${GKE_RELEASE_CHANNEL:-""}
 readonly gke_node_version=${GKE_NODE_VERSION:-}
 readonly gce_region=${GCE_CLUSTER_REGION:-}
 readonly storageclass_files=${STORAGECLASS_FILES:-}
-readonly use_staging_driver=${USE_STAGING_DRIVER:-false}
+readonly use_gke_driver=${USE_STAGING_DRIVER:-false}
```
Comment (saikat-royc) [inline test/run-k8s-integration.sh:29]: we should eventually change USE_STAGING_DRIVER too.

**Comment by saikat-royc** [comment]
the change looks good to me with a minor comment. will lgtm the refreshed PR

File: test/run-k8s-integration.sh (hunk 2 above)
Comment (leiyiz) [reply test/run-k8s-integration.sh:29]: ack.

File: test/k8s-integration/main.go (hunk 1 above)
Comment (leiyiz) [reply test/k8s-integration/main.go:63]: ok. Done.

**Pull request merged**